ngraph/core/src/op/experimental_detectron_roi_feature.cpp

```diff
@@ -0,0 +1,136 @@
+
+#include <algorithm>
+#include <memory>
+#include <utility>
+
+#include "itt.hpp"
+#include "ngraph/attribute_visitor.hpp"
+#include "ngraph/op/experimental_detectron_roi_feature.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v6::ExperimentalDetectronROIFeatureExtractor,
+                       "ExperimentalDetectronROIFeatureExtractor",
+                       6);
+
+op::v6::ExperimentalDetectronROIFeatureExtractor::ExperimentalDetectronROIFeatureExtractor(
+    const OutputVector& args, const Attributes& attrs)
+    : Op(args)
+    , m_attrs(attrs)
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v6::ExperimentalDetectronROIFeatureExtractor::ExperimentalDetectronROIFeatureExtractor(
+    const NodeVector& args, const Attributes& attrs)
+    : ExperimentalDetectronROIFeatureExtractor(as_output_vector(args), attrs)
+{
+}
+
+bool op::v6::ExperimentalDetectronROIFeatureExtractor::visit_attributes(AttributeVisitor& visitor)
+{
+    NGRAPH_OP_SCOPE(v6_ExperimentalDetectronROIFeatureExtractor_visit_attributes);
+    visitor.on_attribute("output_size", m_attrs.output_size);
+    visitor.on_attribute("sampling_ratio", m_attrs.sampling_ratio);
+    visitor.on_attribute("pyramid_scales", m_attrs.pyramid_scales);
+    visitor.on_attribute("aligned", m_attrs.aligned);
+    return true;
+}
+
+void op::v6::ExperimentalDetectronROIFeatureExtractor::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v6_ExperimentalDetectronROIFeatureExtractor_validate_and_infer_types);
+    NODE_VALIDATION_CHECK(this, get_input_size() >= 2, "At least two argument required.");
+
+    auto rois_shape = get_input_partial_shape(0);
+    auto input_et = get_input_element_type(0);
+
+    PartialShape out_shape = {
+        Dimension::dynamic(), Dimension::dynamic(), m_attrs.output_size, m_attrs.output_size};
+
+    PartialShape out_rois_shape = {Dimension::dynamic(), 4};
+    if (rois_shape.rank().is_static())
+    {
+        NODE_VALIDATION_CHECK(
+            this, rois_shape.rank().get_length() == 2, "Input rois rank must be equal to 2.");
+
+        NODE_VALIDATION_CHECK(this,
+                              rois_shape[1].is_static() && rois_shape[1].get_length() == 4u,
+                              "The last dimension of the 'input_rois' input must be equal to 4. "
+                              "Got: ",
+                              rois_shape[1]);
+
+        out_shape[0] = rois_shape[0];
+        out_rois_shape[0] = rois_shape[0];
+    }
+
+    size_t num_of_inputs = get_input_size();
+    std::vector<Dimension> channels(num_of_inputs - 1);
+
+    for (size_t i = 1; i < num_of_inputs; i++)
+    {
+        auto& channel = channels[i - 1];
+        auto current_shape = get_input_partial_shape(i);
+        auto current_rank = current_shape.rank();
+
+        if (current_rank.is_dynamic() || current_shape[1].is_dynamic())
+        {
+            set_output_type(0, input_et, out_shape);
+            return;
+        }
```

Comment:
We should not return from the `validate_and_infer_types` completely if one of the inputs have dynamic rank or channels dimension. See my comment below.

Reply:
Where is this comment?

Reply:
It is for line 95.